In-house patch to fix upstream bug PUP-2509
https://tickets.puppetlabs.com/browse/PUP-2509
Patch has been submitted upstream but not yet accepted.

--- puppet-3.8.6/lib/puppet/provider/service/smf.rb.orig	2016-04-19 14:53:01.233681735 -0700
+++ puppet-3.8.6/lib/puppet/provider/service/smf.rb	2016-04-19 14:53:31.762390677 -0700
@@ -36,7 +36,7 @@
   end
 
   def self.instances
-   svcs.split("\n").select{|l| l !~ /^legacy_run/ }.collect do |line|
+   svcs("-H").split("\n").select{|l| l !~ /^legacy_run/ }.collect do |line|
      state,stime,fmri = line.split(/\s+/)
      status =  case state
                when /online/; :running
